mimxrt: Stop a previous timer on channel reuse.

robert-hh opened the pull request:
As suggested by Damien. Doing it now, before it fades away in the past.

robert-hh committed:
mimxrt: Remove _WFE() from MICROPY_EVENT_POLL_HOOK.

robert-hh committed:
mimxrt: Stop the previous timer on channel reuse.

dpgeorge commented on ports/mimxrt/machine_timer.c:
```diff
@@ -148,6 +148,12 @@ STATIC mp_obj_t machine_timer_make_new(const mp_obj_type_t *type, size_t n_args,
+    if (MP_STATE_PORT(timer_table)[id] != NULL) {
+        PIT_StopTimer(PIT, channel_no[id]);
+        MP_STATE_PORT(timer_table)[id] = NULL;
+    }
```
I'd suggest to just re-use the object:
```c
if (MP_STATE_PORT(timer_table)[id] == NULL) {
    self = m_new_obj_with_finaliser(machine_timer_obj_t);
    MP_STATE_PORT(timer_table)[id] = self;
} else {
    self = MP_STATE_PORT(timer_table)[id];
}
```
That way the hardware timer is mapped 1-1 to an object, and you can retrieve the existing object via `machine.Timer(id)`.  This is how other ports work.

robert-hh committed:
mimxrt: Timer - reuse existing timer object.

robert-hh committed:
mimxrt: Leave the Timer clock source at IPG clock.

Setting it to OSC_CLK interferes time the  utime module.
This is still an area demanding an understanding.

dpgeorge commented:
This looks better now, but I've yet to test it.

Do you also need to remove the WFE from the other locations, or is it enough to remove it from the EVENT_POLL_HOOK only?

robert-hh commented:
The only place where I ever changed it was at the EVENT_POLL_HOOK.

dpgeorge commented:
Merged in 5226d6e1ee7282bf40a45708b98067446b157bce, cdd95ce73705ec8dc5249a7e22092a8ce374576d and d79105d7c0f68e65ff175b752d9895241e355c08 (with a fix so that init can be called again after deinit).
